File: GXpert/GXpert.Web/Modules/Institute/InstituteClass/InstituteClassColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Institute.Columns;

[ColumnsScript("Institute.InstituteClass")]
[BasedOnRow(typeof(InstituteClassRow), CheckNames = true)]
public class InstituteClassColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string ClassTitle { get; set; }
    [EditLink]
    public string Division { get; set; }
    public string MediumTitle { get; set; }
    public string ClassTeacherPrn { get; set; }
    public string Title { get; set; }
    public string AcademicYearName { get; set; }
    public DateTime InsertDate { get; set; }
    public int InsertUserId { get; set; }
    public DateTime UpdateDate { get; set; }
    public int UpdateUserId { get; set; }
    public bool IsActive { get; set; }
    public string InstituteName { get; set; }
    public string DepartmentTitle { get; set; }
    public string CourseTitle { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Institute/InstituteClassColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { InstituteClassRow } from "./InstituteClassRow";

export interface InstituteClassColumns {
    Id: Column<InstituteClassRow>;
    ClassTitle: Column<InstituteClassRow>;
    Division: Column<InstituteClassRow>;
    Semester: Column<InstituteClassRow>;
    ClassTeacherPrn: Column<InstituteClassRow>;
    Title: Column<InstituteClassRow>;
    AcademicYearName: Column<InstituteClassRow>;
    InstituteName: Column<InstituteClassRow>;
    DepartmentTitle: Column<InstituteClassRow>;
    CourseTitle: Column<InstituteClassRow>;
}

export class InstituteClassColumns extends ColumnsBase<InstituteClassRow> {
    static readonly columnsKey = 'Institute.InstituteClass';
    static readonly Fields = fieldsProxy<InstituteClassColumns>();
}